contracts/non-fungible-tokens/andromeda-auction/src/contract.rs

```diff
@@ -648,33 +651,35 @@ fn query_is_claimed(
     env: Env,
     token_id: String,
     token_address: String,
-) -> Result<bool, ContractError> {
+) -> Result<IsClaimedResponse, ContractError> {
     let token_auction_state =
         get_existing_token_auction_state(deps.storage, &token_id, &token_address)?;
 
     let token_owner =
         query_owner_of(deps.querier, token_auction_state.token_address, token_id)?.owner;
 
     // if token owner isn't the contract, it means that it has been claimed. If they're equal it means that it hasn't been claimed and will return false
-    Ok(token_owner != env.contract.address)
+    Ok(IsClaimedResponse {
+        is_claimed: token_owner != env.contract.address,
+    })
 }
 
 fn query_is_closed(
     deps: Deps,
     env: Env,
     token_id: String,
     token_address: String,
-) -> Result<bool, ContractError> {
+) -> Result<IsClosedResponse, ContractError> {
     let token_auction_state =
         get_existing_token_auction_state(deps.storage, &token_id, &token_address)?;
 
-    if query_is_claimed(deps, env.clone(), token_id.clone(), token_address.clone())?
-        || query_is_cancelled(deps, token_id, token_address)?
+    if query_is_claimed(deps, env.clone(), token_id.clone(), token_address.clone())?.is_claimed
```

Comment:
This can probably be restructured to be clearer about what each element in the boolean expression is then return `is_closed` as the relevant outcome of the boolean.